src/datumaro/plugins/ndr.py

```diff
@@ -186,72 +202,137 @@ def __init__(
             unknown_member_error="Unknown undersampling method '{value}'.",
         )
 
+        self._sample_keys = []
+        self._embeddings = []
+        self._deduplicated_item_ids = None
+        
         if seed:
             self.seed = seed
         else:
             self.seed = None
         self.working_subset = working_subset
         self.duplicated_subset = duplicated_subset
         self.algorithm = algorithm
-
         self.num_cut = num_cut
         self.over_sample = over_sample
         self.under_sample = under_sample
-
         self.algorithm_specific = kwargs
+        self.kept_item_id = None
         self._initialized = False
+        self.save_media= save_media
```

Comment:
Space before `=` is missing

Reply:
From my side, that's it. Perhaps @djdameln has something to add
Also, it'd be extremely convenient for us if you provide as small video dataset to check if the plugin is not broken after the refactoring